src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -3108,12 +3108,7 @@ private void RemoveDirectoryInfoItem(DirectoryInfo directory, bool recurse, bool
             {
                 try
                 {
-                    // TODO:
-                    // Different symlinks seem to vary by behavior.
-                    // In particular, OneDrive symlinks won't remove without recurse,
-                    // but the .NET API here does not allow us to distinguish them.
-                    // We may need to revisit using p/Invokes here to get the right behavior
-                    directory.Delete();
+                    directory.Delete(recursive: recurse);
```

Comment:
I'd add a comment that the API correctly works with non named surrogates like a directory on OneDrive. This could help code readers to understand the code path more deeply.

Reply:
Good point, will add a comment why this is needed.